src/lua/api-gateway/validation/validator.lua

```diff
@@ -102,7 +102,7 @@ function BaseValidator:getKeyFromRedis(key, hash_name)
         return self:getHashValueFromRedis(key, hash_name)
     end
 
-    local ok, redisread = redisConnectionProvider:getConnection(self.redis_RO_upstream);
+    local ok, redisread = redisConnectionProvider:getConnection("apiKey", true);
```

Comment:
This is a base validator - the class that extends this one should be able to dictate what type of redis cache to use. In this case you are forcing apiKey all the way - self.redis_RO_upstream was good